proxy/grpcproxy/kv.go

```diff
@@ -35,8 +35,8 @@ func NewKvProxy(c *clientv3.Client) pb.KVServer {
 }
 
 func (p *kvProxy) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
-	// if request set Serializable, serve it from local cache first
-	if r.Serializable {
+	// if request set serializable or a revision is specified, serve it from local cache first
+	if r.Serializable || r.Revision != 0 {
```

Comment:
OK, yeah, otherwise it won't pick up compaction; ignore what I said about it being in the cache. This should still be `if r.Serializable {...}`, though.